### PR TITLE
fix: a few purge_dups tweaks

### PR DESCRIPTION
**Pull request by prototaxites**
Noticed a couple of little points for purge_dups that could be fixed:

- purgedups/purgedups outputs non-compressed BED file but purgedups/getseqs is able to read a compressed BED file, so this is compressed now.
- purgedups/getseqs hard-codes the `-e` option, which is not necessary for operation, but is recommended. Sets this as the default arg in case `ext.args` is not set, but explicitly captures the case that `ext.args` is an empty string and will pass this through in this case, allowing the `-e` option to be unset. 
- Wrap all variables in `${}` cause it's prettier :) 

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Broadcast software version numbers to `topic: versions` - [See version_topics](https://nf-co.re/blog/2025/version_topics)
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/purgedups/getseqs/tests/main.nf.test.snap [hunk 1]
```diff
@@ -7,7 +7,7 @@
                     {
                         "id": "test"
                     },
-                    "test.purged.fa:md5,70d0b87ecc04353767b1a713992f00fe"
+                    "test.purged.fa:md5,5593b561f5dac649dd02adbc1ff440e9"
```
Comment (maxulysse) [inline modules/nf-core/purgedups/getseqs/tests/main.nf.test.snap]: why is the fa md5 changing?

Comment (prototaxites) [reply modules/nf-core/purgedups/getseqs/tests/main.nf.test.snap]: Removing the `-e` option changes the operation of the tool

File: modules/nf-core/purgedups/getseqs/main.nf [hunk 2]
```diff
@@ -24,10 +24,10 @@ process PURGEDUPS_GETSEQS {
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     get_seqs \\
-        $args \\
-        -e $bed \\
```
Comment (PeterKneale) [inline modules/nf-core/purgedups/getseqs/main.nf:28]: given the below: do you think it would make sense to have the `-e` parameter set as a default if no other configuration is supplied? would that be a safer transition for people already consuming this module?

Image

Comment (prototaxites) [reply modules/nf-core/purgedups/getseqs/main.nf:28]: Yes, that's probably a good idea

Comment (prototaxites) [reply modules/nf-core/purgedups/getseqs/main.nf:28]: OK, fixed. I explicitly check if task.ext.args is a string, and if so we use it even if it's empty. Otherwise we fall back to `-e`, catching the case where it is unset and thus null.

**Comment by prototaxites** [comment]
@nf-core-bot fix linting

File: modules/nf-core/purgedups/getseqs/main.nf [hunk 3]
```diff
@@ -20,14 +20,16 @@ process PURGEDUPS_GETSEQS {
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
+    // By default, we include the '-e' option in task.ext.args as this is considered the 'safe default'
```
Comment (prototaxites) [inline modules/nf-core/purgedups/getseqs/main.nf:23]:
```suggestion
    // By default, we include the '-e' option in task.ext.args as this is considered the 'safe default'
    // Check if we have a CharSequence (both String and GString inherit) and thus we can apply an empty string, which is otherwise 'falsy'
```

**Comment by prototaxites** [comment]
Up to you, but FYI any nf-core member is welcome to review and approve code :) 

See https://nf-co.re/docs/contributing/how_to_contribute_to_nf-core

File: modules/nf-core/purgedups/getseqs/tests/main.nf.test [hunk 4]
```diff
@@ -22,6 +23,36 @@ nextflow_process {
...
+    test("sarcov2 - fasta - bed - no e option") {
+
+        // load a config file where we load ext.args as an empty string to unset -e option
+        config './nextflow.config'
+
+        when {
+            process {
+                """
+                ch_bed = Channel.of('MT192765.1\\n0\\t29828\\tJUNK\\n').collectFile(name: 'test.bed' )
+                input[0] = Channel.of([
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/getseqs/tests/main.nf.test]:
```suggestion
                ch_bed = channel.of('MT192765.1\\n0\\t29828\\tJUNK\\n').collectFile(name: 'test.bed' )
                input[0] = channel.of([
```
Let's try to keep them like that throughout already

File: modules/nf-core/purgedups/purgedups/main.nf [hunk 5]
```diff
@@ -24,16 +24,18 @@ process PURGEDUPS_PURGEDUPS {
...
+        -c ${basecov} \\
+        ${paf} | gzip \\
+        > ${prefix}.dups.bed.gz \\
+        2>| >(tee ${prefix}.purge_dups.log >&2)
     """
 
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}.dups.bed
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/purgedups/main.nf:36]: echo the args as well

File: modules/nf-core/purgedups/purgedups/main.nf [hunk 6]
```diff
@@ -24,16 +24,18 @@ process PURGEDUPS_PURGEDUPS {
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     purge_dups \\
-        $args \\
-        -T $cutoff \\
-        -c $basecov \\
-        $paf > ${prefix}.dups.bed 2>| >(tee ${prefix}.purge_dups.log >&2)
+        ${args} \\
+        -T ${cutoff} \\
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/purgedups/main.nf:28]: is this a mandatory input? else remove and let it be provided by ext.args

File: modules/nf-core/purgedups/purgedups/main.nf [hunk 7]
```diff
@@ -24,16 +24,18 @@ process PURGEDUPS_PURGEDUPS {
...
+        -c ${basecov} \\
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/purgedups/main.nf:29]: same for this (cant see inputs here to see if this is a file)

File: modules/nf-core/purgedups/histplot/main.nf [hunk 8]
```diff
@@ -27,9 +27,9 @@ process PURGEDUPS_HISTPLOT {
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     hist_plot.py \\
-        -c $cutoff \\
-        $args \\
-        $statfile \\
+        -c ${cutoff} \\
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/histplot/main.nf]: If this is not mandatory, provide through ext.args instead

File: modules/nf-core/purgedups/getseqs/tests/main.nf.test [hunk 9]
```diff
@@ -22,6 +23,36 @@ nextflow_process {
...
+                    [ id:'test'],
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
+                ]).combine(ch_bed)
+                """
+            }
+        }
+
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/getseqs/tests/main.nf.test:88]: Maybe also add a test with a config that passes somethign else rather than empty or -e? Else it is not obvious how this can get overriden

File: modules/nf-core/purgedups/purgedups/main.nf (hunk 6 above)
Comment (prototaxites) [reply modules/nf-core/purgedups/purgedups/main.nf:28]: I believe these are mandatory for this one

File: modules/nf-core/purgedups/histplot/main.nf (hunk 8 above)
Comment (prototaxites) [reply modules/nf-core/purgedups/histplot/main.nf]: done

**Comment by prototaxites** [comment]
test with args unset (`null`):
```
#!/bin/bash -ue
get_seqs \
    -e \
    test.bed \
    -p test \
    genome.fasta
```

test with args set to empty string:
```
#!/bin/bash -ue
get_seqs \
     \
    test.bed \
    -p test \
    genome.fasta
```

test with args set explicitly to another string, `-es`:
```
#!/bin/bash -ue
get_seqs \
    -es \
    test.bed \
    -p test \
    genome.fasta
```

File: modules/nf-core/purgedups/getseqs/tests/main.nf.test (hunk 9 above)
Comment (prototaxites) [reply modules/nf-core/purgedups/getseqs/tests/main.nf.test:88]: I have added an extra test that adds some explicit args, to demonstrate that setting args always overrides the default `-e`.

File: modules/nf-core/purgedups/getseqs/tests/main.nf.test [hunk 10]
```diff
@@ -10,18 +10,90 @@ nextflow_process {
...
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    file(process.out.haplotigs[0][1]).name, // Empty file
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/getseqs/tests/main.nf.test]: hmm when is this non-empty then? does it need another test maybe?

File: modules/nf-core/purgedups/purgedups/meta.yml [hunk 11]
```diff
@@ -44,10 +44,10 @@ output:
           description: |
             Groovy Map containing sample information
             e.g. [ id:'test', single_end:false ]
-      - "*.dups.bed":
+      - "*.dups.bed.gz":
           type: file
           description: A bed file of sequences purged of false duplications
-          pattern: "*.bed"
+          pattern: "*.bed.gz"
           ontologies: []
```
Comment (vagkaratzas) [inline modules/nf-core/purgedups/purgedups/meta.yml]: this can probably be fille dbefore merging

File: modules/nf-core/purgedups/getseqs/tests/main.nf.test (hunk 10 above)
Comment (prototaxites) [reply modules/nf-core/purgedups/getseqs/tests/main.nf.test]: It's non-empty in real cases where there are haplotigs actually being purged from a genome assembly. I'm not sure how to make this work in this test case...

Comment (prototaxites) [reply modules/nf-core/purgedups/getseqs/tests/main.nf.test]: The test BED file was broken... fixed and the file is not empty!

File: modules/nf-core/purgedups/purgedups/meta.yml (hunk 11 above)
Comment (prototaxites) [reply modules/nf-core/purgedups/purgedups/meta.yml]: Done!
